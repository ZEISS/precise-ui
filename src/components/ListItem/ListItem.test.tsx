import * as React from 'react';
import * as enzyme from 'enzyme';
import { ListItem } from './';

describe('<ListItem />', () => {
  it('should render styled.li as default', () => {
    const wrapper = enzyme.shallow(<ListItem />);
    expect(wrapper.name()).toBe('styled.li');
  });

  it('should be render with padding, by default', () => {
    const wrapper = enzyme.mount(<ListItem />);
    expect(wrapper).toHaveStyleRule('padding', '0.75rem 1.25rem');
  });

  it('should render component without padding', () => {
    const wrapper = enzyme.mount(<ListItem disablePadding />);
    expect(wrapper).toHaveStyleRule('padding', '0');
  });

  it('should not present a border', () => {
    const wrapper = enzyme.mount(<ListItem border={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
