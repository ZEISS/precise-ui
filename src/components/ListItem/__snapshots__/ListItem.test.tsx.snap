// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should not present a border 1`] = `
.c0 {
  background-color: rgba(255,255,255,1);
  color: rgba(0,0,0,1);
  margin-bottom: -1px;
  padding: 0.75rem 1.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  margin-bottom: 0;
}

<ListItem
  border={false}
>
  <styled.li
    border={false}
  >
    <li
      className="c0"
    />
  </styled.li>
</ListItem>
`;
